pycalphad/core/utils.py

```diff
@@ -428,3 +428,102 @@ def wrap_symbol(obj):
     else:
         return Symbol(obj)
 
+
+def recursive_tuplify(x):
```

Comment:
Duplicate of https://github.com/pycalphad/pycalphad/blob/56201bfb10372fb7ef90efde4f3fa431781c36bc/pycalphad/io/database.py#L20-L22
We should pick one to keep and remove the other.

Reply:
Going to keep `recursive_tuplify` as it will descend into tuples and tuplify them, for example:

`[ ([], []), ([], []) ]`:

```python
>>> _to_tuple( [ ([], []), ([], []) ] )
(([], []), ([], []))

>>> recursive_tuplify( [ ([], []), ([], []) ] ) 
(((), ()), ((), ()))
```

For the current use case of `_to_tuple` (only in `Database.__hash__` for model hints), `recursive_tuplify` should be more correct.